src/figuresClasses.ts

```diff
@@ -1,11 +1,88 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Side lengths must be positive numbers');
+    }
+
+    const arrOfSides = [a, b, c];
+    const maxSide = Math.max(...arrOfSides);
+    const smallerSidesSum = arrOfSides
+      .filter((item) => item < maxSide)
+      .reduce((acc, currentValue) => acc + currentValue);
```

Comment:
The logic for checking the triangle inequality is incorrect. The current implementation only checks if the largest side is less than the sum of the other two sides. However, the triangle inequality theorem states that the sum of the lengths of any two sides must be greater than the length of the third side for all three combinations of sides.